extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyRowFilterAndDataMasking.scala

```diff
@@ -30,29 +30,40 @@ import org.apache.kyuubi.plugin.spark.authz.util.RowFilterAndDataMaskingMarker
 class RuleApplyRowFilterAndDataMasking(spark: SparkSession) extends Rule[LogicalPlan] {
 
   override def apply(plan: LogicalPlan): LogicalPlan = {
-    // Apply FilterAndMasking and wrap HiveTableRelation/LogicalRelation with
+    // Apply FilterAndMasking and wrap HiveTableRelation/LogicalRelation/DataSourceV2Relation with
     // RowFilterAndDataMaskingMarker if it is not wrapped yet.
     plan mapChildren {
       case p: RowFilterAndDataMaskingMarker => p
       case hiveTableRelation if hasResolvedHiveTable(hiveTableRelation) =>
         val table = getHiveTable(hiveTableRelation)
-        applyFilterAndMasking(hiveTableRelation, table, spark)
+        applyFilterAndMasking(hiveTableRelation, table.identifier, spark)
       case logicalRelation if hasResolvedDatasourceTable(logicalRelation) =>
         val table = getDatasourceTable(logicalRelation)
         if (table.isEmpty) {
           logicalRelation
         } else {
-          applyFilterAndMasking(logicalRelation, table.get, spark)
+          applyFilterAndMasking(logicalRelation, table.get.identifier, spark)
+        }
+      case datasourceV2Relation if hasResolvedDatasourceV2Table(datasourceV2Relation) =>
+        val table = getDatasourceV2Table(datasourceV2Relation)
+        if (table == null) {
+          datasourceV2Relation
+        } else {
+          val catalogDbTable = table.name.split("\\.")
```

Comment:
```java
  /**
   * A name to identify this table. Implementations should provide a meaningful name, like the
   * database and table name from catalog, or the location of files for this table.
   */
```

Check the comments above from the spark codebase, the cases below seem not to be handled,

1. the location of files for this table
2. when the catalog name is absent
3. backtick-quoted name parts, such as "cat1.db.`abc.xyz`

Reply:
I have also noticed the shortcomings. The current implementation is not elegant as espected. 
In `org.apache.spark.sql.connector.catalog.Table`, `name()` method is not providing enough infomation to identify all these situations.

Reply:
I wil have more investigation in `org.apache.spark.sql.connector.catalog.Identifier` in `DataSourceV2Relation` to see whether it is a possible path for this problem.

Reply:
Changed to rely on  `org.apache.spark.sql.connector.catalog.Identifier` of `DataSourceV2Relation` instead of unconvincing name() in`Table`.

Resolving the previously listed problems by: 
1. using the name of Identifier as accurate table name
2. concating namespaces in Identifier as the full database name